docs/api/web-contents.md

```diff
@@ -207,6 +207,30 @@ this purpose.
 
 Calling `event.preventDefault()` will prevent the navigation.
 
+#### Event: 'will-frame-navigate'
+
+Returns:
+
+* `event` Event
+* `url` String
+* `details` Object
+  * `isMainFrame` boolean
+  * `frame` WebFrameMain
+
+Emitted when a user or the page wants to start navigation in any frame. It can happen when
+the `window.location` object is changed or a user clicks a link in the page.
+
+Unlike `will-navigate`, `will-frame-navigate` is fired when the main frame or any of its subframes attempts to navigate. When the navigation event comes from the main frame, `isMainFrame` will be `true`.
+
+This event will not emit when the navigation is started programmatically with
+APIs like `webContents.loadURL` and `webContents.back`.
+
+It is also not emitted for in-page navigations, such as clicking anchor links
+or updating the `window.location.hash`. Use `did-navigate-in-page` event for
+this purpose.
```

Comment:
I feel like an app developer is going to need to know about & understand the nuances between `did-navigate-in-page`, `will-navigate`, and `will-frame-navigate` now. Maybe there should be a standalone section that discusses this so that all  three events can link to that in the documentation?